docs/tutorials/remote_inference.md

```diff
@@ -0,0 +1,476 @@
+```
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "sagemaker: embedding",
+    "description": "Test connector for Sagemaker embedding model",
+    "version": 1,
+    "protocol": "aws_sigv4",
+    "credential": {
+        "access_key": "<Your AWS account access key>",
+        "secret_key": "<Your AWS account secret key>",
+        "session_token": "<Your AWS account session token>"
+    },
+    "parameters": {
+        "region": "<Your Amazon Sagemaker Region, for example us-west-2>",
+        "service_name": "sagemaker"
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "headers": {
+                "content-type": "application/json"
+            },
+            "url": "<Your Sagemaker inference endpoint>",
+            "request_body": "<Your Sagemaker model model input, for example {\"input\": \"${parameters.inputText}\"}>"
+        }
+    ]
+}
+```
+## http connector
+
+No reserved parameter/credential names. 
+
+Example to create an OpenAI Chat model connector:
+```
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "OpenAI Chat Connector",
+    "description": "The connector to public OpenAI model service for GPT 3.5",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+        "endpoint": "api.openai.com",
+        "model": "gpt-3.5-turbo"
+    },
+    "credential": {
+        "openAI_key": "<Your OpenAI API key>"
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "url": "https://${parameters.endpoint}/v1/chat/completions",
+            "headers": {
+                "Authorization": "Bearer ${credential.openAI_key}"
+            },
+            "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": ${parameters.messages} }"
+        }
+    ]
+}
+```
+
+
+# Remote model
+
+User can define a remote model with connector. User have two ways to create a remote model
+
+- Remote model with connector id
+- Remote model with an internal connector
+
+Suggest you create a model group first, it will be used by other APIs later.
+
+Sample request to create model group:
+```
+POST /_plugins/_ml/model_groups/_register
+{
+  "name": "my_remote_model_group",
+  "description": "This is an example description"
+}
+```
+Sample response
+```
+{
+  "model_group_id": "WSmckYkBrQ7TyjgDAhjb",
+  "status": "CREATED"
+}
+```
+
+## Remote model with connector id
+
+User can run `POST /_plugins/_ml/connectors/_create` to register a connector first. Then use the connector id to create a remote model.
+
+### Step1: Create connector
+Sample request: create OpenAI Chat model connector
+```
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "OpenAI Chat Connector",
+    "description": "The connector to public OpenAI model service for GPT 3.5",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+        "model": "gpt-3.5-turbo"
+    },
+    "credential": {
```

Comment:
We should state that this credential provided to Ml-commons will not be shown to the customer through any APIs, so it's customers' responsibility to store the credential safely.

Reply:
Thanks, will address the comments in a new PR soon.